### PR TITLE
feat: add configurable weapon sfx to enemies

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Comment by ghost** [comment]
This should be Ready come morning after replays finish. time for bed.

File: src/base/qrs.h [hunk 1]
```diff
@@ -214,7 +214,7 @@ enum
 	qr_OLD_SLASHNEXT_SECRETS, qr_STEP_IS_FLOAT, qr_OLDMAPDATAFLAGS, qr_OLD_PRINTF_ARGS,
 	qr_PASSIVE_SUBSCRIPT_RUNS_WHEN_GAME_IS_FROZEN, qr_WRITE_ENTRYPOINTS_AFFECTS_HEROCLASS, qr_LOG_INVALID_UID_LOAD, qr_COMBODATA_INITD_MULT_TENK,
 	//86
-	qr_ZS_NO_NEG_ARRAY, qr_PRINTF_NO_0FILL, qr_SHIELDBUBBLE,
+	qr_ZS_NO_NEG_ARRAY, qr_PRINTF_NO_0FILL, qr_SHIELDBUBBLE, qr_SETENEMYFIRESFXONWPNCHANGE,
```
Comment (connorjclark) [inline src/base/qrs.h]: this qr is not used anywhere, remove?

File: src/zc/guys.cpp [hunk 2]
```diff
@@ -2007,7 +2010,10 @@ void enemy::FireWeapon()
 			
 			if(summoned)
 			{
-				sfx(get_qr(qr_MORESOUNDS) ? WAV_ZN1SUMMON : WAV_FIRE,pan(int32_t(x)));
+				if (replay_version_check(0, 35))
```
Comment (connorjclark) [inline src/zc/guys.cpp]: If the qst loading compat code that sets firesfx is working correctly ... I'd expect this replay check / version bump to be unnecessary.

Comment (Jambu1989) [reply src/zc/guys.cpp:2013]: wizzrobes depend on it.

Comment (ghost) [reply src/zc/guys.cpp]: Yeah one would expect this... Unfortunately...
https://github.com/Jambu1989/ZQuestClassic/actions/runs/10582185604/job/29321832440

**Comment by ghost** [comment]
should be ready to merge now

File: src/zc/guys.cpp (hunk 2 above)
Comment (connorjclark) [reply src/zc/guys.cpp]: The issue was that quests with old guy sections were not running this firesfx compat code, and the default_guys all had these fields set to zero.

I pushed a change to resolve that.

File: resources/docs/ZScript_Additions.txt [hunk 3]
```diff
@@ -5799,6 +5799,9 @@ untyped Attributes[32];
 int WeaponSprite;
  * The sprite (Quest->Graphics->Sprites->Weapons) used to draw the weapon fired by the npc.
 
+int WeaponSFX;
+ * The sfx (Quest->Audio->SFXData) used whenever the npc fires it's weapon.
```
Comment (connorjclark) [inline resources/docs/ZScript_Additions.txt]:
```suggestion
 * The sfx (Quest->Audio->SFXData) used whenever the npc fires its weapon.
```

File: resources/docs/ZScript_Additions.txt [hunk 4]
```diff
@@ -8077,6 +8080,9 @@ int TileHeight;
 int WeaponSprite;
  * The sprite used to draw the enemy weapon.
 
+ int WeaponSFX;
+ * The sfx used when the enemy fires it's weapon.
```
Comment (connorjclark) [inline resources/docs/ZScript_Additions.txt]:
```suggestion
 * The sfx used when the enemy fires its weapon.
```

File: resources/include/bindings/npc.zh [hunk 5]
```diff
@@ -573,6 +573,12 @@ class npc {
 	// @zasm_var NPCWEAPSPRITE
 	internal int WeaponSprite;
 
+	// The sfx (Quest->Audio->SFXData) used whenever the npc fires it's weapon.
```
Comment (connorjclark) [inline resources/include/bindings/npc.zh]:
```suggestion
	// The sfx (Quest->Audio->SFXData) used whenever the npc fires its weapon.
```

File: resources/include/bindings/npcdata.zh [hunk 6]
```diff
@@ -227,6 +227,12 @@ class npcdata {
 	// @zasm_var NPCDATAWPNSPRITE
 	internal int WeaponSprite;
 
+	// The sfx used when the enemy fires it's weapon.
```
Comment (connorjclark) [inline resources/include/bindings/npcdata.zh]:
```suggestion
	// The sfx used when the enemy fires its weapon.
```

File: src/zc/scripting/types/npc.cpp [hunk 7]
```diff
@@ -1990,7 +1993,11 @@ bool npc_set_register(int32_t reg, int32_t value)
...
 				GuyH::getNPC()->wpn = weapon;
 			
-				if ( get_qr(qr_SETENEMYWEAPONSPRITESONWPNCHANGE) ) //this should probably just be an extra_rule
+				if ( get_qr(qr_SETENEMYWEAPONSPRITESONWPNCHANGE) )
+				{
+					GuyH::getNPC()->wpnsprite = FFCore.GetDefaultWeaponSprite(weapon);
+				}
+				if (get_qr(qr_SETENEMYWEAPONSOUNDSONWPNCHANGE))
 				{
 					GuyH::getNPC()->wpnsprite = FFCore.GetDefaultWeaponSprite(weapon);
```
Comment (connorjclark) [inline src/zc/scripting/types/npc.cpp]: this is setting the sprite, not the sound.

File: src/zc/scripting/types/npc.cpp [hunk 8]
```diff
@@ -2433,6 +2440,13 @@ bool npc_set_register(int32_t reg, int32_t value)
 			}
 			break;
 
+		case NPCFIRESFX:
+			if (GuyH::loadNPC(ri->guyref, "npc->WeaponSFX") == SH::_NoError)
+			{
+				GuyH::getNPC()->firesfx = vbound(value / 10000, -1, 255);
```
Comment (connorjclark) [inline src/zc/scripting/types/npc.cpp]:
```suggestion
				GuyH::getNPC()->firesfx = vbound(value / 10000, 0, 255);
```

might as well be 0?

File: src/zc/scripting/types/npc.cpp (hunk 7 above)
Comment (ghost) [reply src/zc/scripting/types/npc.cpp]: doh

**Review by connorjclark — APPROVED** [review]

**Pull request merged**